dlt/destinations/weaviate/weaviate_client.py

```diff
@@ -454,37 +456,91 @@ def get_storage_table(self, table_name: str) -> Tuple[bool, TTableSchemaColumns]
             table_schema[prop["name"]] = schema_c
         return True, table_schema
 
-    def get_schema_by_hash(self, schema_hash: str) -> Optional[StorageSchemaInfo]:
-        version_class_name = self.schema.version_table_name
-
-        try:
-            self.get_class_schema(version_class_name)
-        except weaviate.exceptions.UnexpectedStatusCodeException as e:
-            if e.status_code == 404:
+    def get_stored_state(self, state_table: str, pipeline_name: str) -> Optional[str]:
+        """Loads compressed state from destination storage"""
+
+        # we need to find a stored state that matches a load id that was completed
+        # we retrieve the state in blocks of 10 for this
+        stepsize = 10
+        offset = 0
+        while True:
+            state_records = self.get_records(state_table,
+                sort={
+                    "path": ["created_at"],
+                    "order": "desc"
+                }, where={
+                    "path": ["pipeline_name"],
+                    "operator": "Equal",
+                    "valueString": pipeline_name,
+                }, limit=stepsize, offset=offset)
+            offset += stepsize
+            if len(state_records) == 0:
                 return None
-            raise
-
-        properties = list(self.schema.get_table_columns(version_class_name).keys())
+            for state in state_records:
+                load_id = state["_dlt_load_id"]
+                load_records = self.get_records(self.schema.loads_table_name,
+                     where={
+                        "path": ["load_id"],
+                        "operator": "Equal",
+                        "valueString": load_id,
+                     }, limit=1)
+                # if there is a load for this state which was successfull, return the state
+                if len(load_records) and load_records[0]["status"] == 0:
```

Comment:
theoretically this will not work if there's transformation layer that adds more statuses ie. 1,2 etc. jus tthe presence of load is sufficient. don't check the status

Reply:
done